## Introduction

Maintaining genome integrity in the mammalian germline is enormously complex. Hundreds of protein-coding genes contribute to pathways involved in DNA replication, and hundreds more are mobilized in response to damage by exogenous and endogenous mutagens. Despite this abundance of potential targets, *mutator alleles* that augment the germline mutation rate have largely eluded detection in mammals. 

Germline mutator alleles are difficult to detect for a number of reasons, including the fidelity of germline genome replication and the effects of selection on mutators. On average, humans are born with about 70 to 100 *de novo* germline mutations per diploid genome [@PMID:28959963; @PMID:31549960]; in mice, the number is closer to 20 or 30 [@PMID:31492841]. Moreover, in a population of sufficiently large $N_e$, we would also expect even low-effect mutator alleles to be efficiently selected against. The selection coefficient on a mutator allele is approximately $2s \Delta U$ [@PMID:27739533], where *s* is the mean selective coefficient on a new deleterious mutation and $\Delta U$ is the excess number of new deleterious mutations caused by the mutator allele; the product of $s$ and $\Delta U$ is multiplied by 2 to account for the expected number of generations for which mutator will be linked to the excess mutations it causes. Given the low germline *de novo* mutation rate in mamalian genomes and the strength of selection on a potential mutator allele, we would likely require a very large number of offspring, as well as an environment that attenuates theeffects of selection, in order to detect the effects of a germline mutator allele.

*de novo* germline mutations are Protein-coding genes involved in DNA replication and repair often recognize particular sequence motifs or excise lesions at specific nucleotides. Thus, we might expect that the spectrum of de novo mutations -- i.e, the frequency of each individual mutation type (C>T, A>G, etc.) -- will differ between genomes that harbor either a mutator or wild-type allele at a given locus.

Previously, we discovered a germline mutator allele in mice by analyzing whole-genome sequencing data from 152 recombinant inbred lines (RILs). These RILs, known as the BXDs, were derived from C57BL/6J and DBA/2J, two laboratory strains that exhibit significant differences in their germline mutation spectra. The BXDs were inbred for up to 150 generations, and each line therefore accumulated up to 2,000 germline de novo mutations. Using quantitative trait locus (QTL) mapping, we identified a locus on chromosome 4 that was strongly associated with the C>A germline mutation rate. The QTL overlapped Mutyh, which encodes a protein that normally prevents C>A mutations by repairing oxidative DNA damage.

In this study, we developed a new method to detect alleles that affect the mutation spectrum in two-parent RILs. At every informative marker along the genome, we compute the aggregate de novo mutation spectrum in lines that inherited either parental allele. We then calculate the cosine distance between those aggregate spectra and identify loci at which that distance is greater than what we'd expect by random chance. By applying this method to mutation data from the BXDs, we discovered an additional C>A germline mutator locus on chromosome 6 that overlaps Ogg1, a key partner of Mutyh in base-excision repair of oxidative DNA damage. BXDs with the mutator allele near Ogg1 do not exhibit elevated rates of C>A germline mutation unless they also possess the mutator allele near Mutyh, but BXDs with both alleles exhibit even higher C>A mutation rates than those with either one alone. 

To our knowledge, these results reveal evidence of epistasis between mammalian germline mutator alleles for the first time. We also demonstrate that new methods for analyzing mutation spectra can identify previously undetected mutator alleles in mammalian genomes, and may enable similar discoveries in humans and other model organisms.
