## Introduction

The germline mutation rate is a fundamental parameter in population genetics, and reflects the complex interplay between DNA replication and repair pathways, exogenous sources of DNA damage, and life-history traits. 
For example, parental age is an important determinant of mutation rate variability; in many mammalian species, the number of germline *de novo* mutations observed in offspring increases as a function of paternal and maternal age [@PMID:28959963;@PMID:31549960;@PMID:35771663;@PMID:32804933;@PMID:31492841].
Rates of germline mutation accumulation are also variable across human families [@PMID:26656846;@PMID:31549960], potentially due to genetic variation or differences in environmental exposures.
Nonetheless, genetic variants that augment germline mutation rates, known as *mutator alleles*, have largely eluded detection in mammals.

Studies of thousands of human trios have failed to identify common variants associated with germline mutation rate variation [@PMID:31964835], meaning that any mutators that segregate within the human population are likely rare or modest in effect size.
The dearth of observed germline mutators in mammalian genomes is not terribly surprising, since alleles that lead to elevated germline mutation rates would likely have deleterious consequences and be purged by negative selection [@PMID:27739533].
Moreover, germline mutation rates are relatively low, and direct mutation rate measurements generally require whole-genome sequencing data from both parents and their offspring.
As a result, large-scale association studies &mdash; which have been used to map genetic contributions to numerous other complex traits &mdash; are not well-powered to investigate the polygenic architecture of germline mutation rates.  

Despite these challenges, less traditional strategies have been successfully used to identify a small number of mutator alleles in both humans and mice. 
By focusing on children who suffer from rare genetic diseases, a recent study identified two mutator alleles that lead to significantly elevated rates of *de novo* germline mutation in human families [@PMID:35545669]. 
Another group discovered mutator phenotypes in the sperm and somatic tissues of adults that suffer from cancers caused by inherited mutations in the POLE/POLD1 exonucleases [@PMID:34594041].

In mice, candidate mutator alleles have also been discovered by sequencing a large collection of recombinant inbred lines (RILs).
Commonly known as the <u>B</u>X<u>D</u>s [@PMID:33472028], these RILs were derived from either F2 or advanced intercrosses of C57<u>B</u>L/6J and <u>D</u>BA/2J, two laboratory strains that exhibit significant differences in their germline mutation spectra [@PMID:30753674]. 
The BXD RILs were maintained via brother-sister mating for up to 180 generations, and each line therefore accumulated hundreds or thousands of germline mutations on a nearly-homozygous linear mosaic of parental <u>B</u> and <u>D</u> haplotypes. 
Due to their husbandry in a controlled laboratory setting, the BXDs were largely free from confounding by environmental heterogeneity, and the effects of selection on *de novo* mutations were attenuated by strict inbreeding [@doi:10.1146/annurev.ecolsys.39.110707.173437]. 

Whole-genome sequencing data from 152 BXD strains were previously used to map a quantitative trait locus (QTL) for the C>A germline *de novo* mutation rate.
Germline C>A mutation rates were nearly 50% higher in mice with *D* alleles at the QTL, likely due to the reduced efficacy of oxidative DNA damage repair caused by genetic variation the DNA glycosylase *Mutyh*.
Importantly, the quantitative trait locus was undetectable in a genome-wide scan for variation in overall mutation rates, which were only modestly higher in BXDs with *D* alleles, demonstrating the utility of *mutation spectrum* analysis for mutator allele discovery.
Close examination of the mutation spectrum is likely to be broadly useful for detecting mutator alleles, as genes involved in DNA replication and repair often recognize particular sequence motifs or excise specific types of DNA lesions [@PMID:32619789].
Mutation spectra are usually defined in terms of $k$-mer nucleotide context [@PMID:32619789]; the 1-mer mutation spectrum, for example, consists of 6 mutation types after collapsing by strand complement (C>T, C>A, C>G, A>T, A>C, A>G), while the 3-mer mutation spectrum contains 96 (each of the 1-mer mutations partitioned by trinucleotide context).

Although mutation spectrum analysis can enable the discovery of mutator alleles that affect the rates of specific mutation types, early implementations of this strategy have suffered from some drawbacks. 
For example, performing association tests on the rates or fractions of every $k$-mer mutation type can quickly incur a substantial multiple testing burden.
Since germline mutation rates are generally quite low, estimates of $k$-mer mutation type frequencies from individual samples may also be noisy and imprecise.
We were therefore motivated to develop a statistical method that could overcome the sparsity of *de novo* mutation spectra, eliminate the need to test each $k$-mer mutation type separately, and enable sensitive detection of alleles that influence the germline mutation spectrum.

Here, we present a new mutation spectrum association test that minimizes multiple testing burden and mitigates the challenges of sparsity in *de novo* mutation datasets. 
We leverage this method to re-analyze germline mutation data from the BXD cohort, and find compelling evidence for the existence of a second mutator allele that was undetectable using previous approaches. 
The new allele appears to interact epistatically with the mutator that was previously discovered in the BXDs, further augmenting the C>A germline mutation rate in a subset of RILs. 
Our observation of epistasis suggests that mild DNA repair deficiencies may compound one another, as mutator alleles chip away at the redundant DNA repair systems that collectively maintain germline intergrity. 
