## Results

### Summary of *de novo* germline mutation data in the BXD RILs

The BXD resource currently comprises a total 152 recombinant inbred lines (RILs). RILs were derived from either F2 or advanced intercross lines (AILs), and subsequently inbred by brother-sister mating for up to 180 generations. Previously, we analyzed whole-genome sequencing data from the BXDs and identified high-quality *de novo* germline mutations in each line. A detailed description of the methods used for variant processing and filtering, as well as the characteristics of the *de novo* mutations, are available in a previous manuscript [@PMID:35545679], but a brief summary follows.

We identified variants in each BXD that were absent from all other RILs, as well as from the C57BL/6J and DBA/2J parents (**Methods**). Since the BXDs were inbred for up to 180 generations, we required each private variant to be homozygous for the private allele; we also included heterozygous variants for which at least 90% of sequencing reads supported the private allele. Counts of "homozygous singletons" were positively correlated with the duration of inbreeding, and as expected for *de novo* germline mutations that have not been purged by selection, singletons were enriched in conserved regions of the genome [@PMID:35545679].

### A new approach to discover germline mutator alleles

We developed a new approach to discover loci that affect the germline *de novo* mutation spectrum in biparental RILs. To illustrate the approach, we provide a toy example below.

First, imagine a population of 4 haplotypes ($h_1$ through $h_4$), which have been genotyped at three informative markers ($g_1$ through $g_3$). Each of these haplotypes also carries a number of *de novo* germline mutations. 

To detect loci that influence the germline mutation spectrum, we can iterate over each informative marker and divide the haplotypes into two groups based on the parental allele they inherited. We then compute a $k$-mer mutation spectrum using the aggregate mutation counts in each haplotype group. The $k$-mer mutation spectrum contains the frequency of every possible $k$-mer mutation type in a collection of mutations, and can be represented as a vector of size $2 \times 3 \times 4^{k - 1}$ after collapsing by strand complement. 

At each locus, we then compute the cosine distance between the aggregate mutation spectra of haplotypes with either parental allele. The cosine distance between two vectors $\mathbf{A}$ and $\mathbf{B}$ is defined as $D_C = 1 - \frac{\mathbf{A} \cdot \mathbf{B}}{||\mathbf{A}|| \ ||\mathbf{B}||}$, and can be viewed as a measure of dissimilarity between two mutation spectra. Since cosine distance does not take the magnitude of vectors into account, it can also be used to compare two spectra with unequal total mutation counts. Additionally, since we calculate the cosine distance between mutation *spectra*, we avoid the need to perform separate comparisons of mutation counts at each individual $k$-mer mutation type.

To assess the significance of cosine distances at particular loci, we use a permutation test to establish genome-wide distance thresholds at a given $p$-value. In each of $N$ permutation trials, we randomly shuffle the individual haplotype mutation data such that haplotype labels no longer correspond to the correct mutation counts. Using the shuffled mutation data, we perform a genome-wide distance scan as described above, and record the maximum distance observed at any locus. After $N$ permutations (usually 1,000 or 10,000), we then compute the $1 - p$ percentile of the maximum distance distribution, and use that percentile value as a genome-wide significance threshold.

### Simulations to assess the power of the inter-haplotype distance approach

### Re-identifying the mutator allele on chromosome 4 in the BXDs

We first applied our inter-haplotype distance method to 94 of the BXD RILs that had been inbred for at least 20 generations, using the previously described *de novo* germline mutation data and a collection of approximately 7,500 genotype markers. Reassuringly, we observed a large increase in cosine distance at a locus on chromosome 4, with a peak cosine distance of X at marker ID rsYYYYY (116.8 Mbp in mm10 coordinates). In a previous analysis, we used quantitative trait locus (QTL) mapping to identify a nearly identical locus on chromosome 4 that was significantly associated with the C>A germline mutation rate in the BXDs. This locus overlaps 21 protein-coding genes that are annotated by the Gene Ontology as being involved in "DNA repair," but only one of these genes contains non-synonymous differences between the two parental strains: *Mutyh*. *Mutyh* encodes a protein involved in the base-excision repair of 8-oxoguanine (8-oxoG), a DNA lesion caused by oxidative damage, and prevents the accumulation of C>A mutations following DNA replication through an 8-oxoG modified base. As expected, C>A germline mutation rates are nearly 50% higher in BXDs that inherited *D* haplotypes at marker ID rsYYYY than in those that inherited *B* haplotypes.

### An additional germline mutator allele on chromosome 6

After confirming that our method recovered the previously-discovered mutator locus overlapping *Mutyh*, we then asked if the inter-haplotype distance approach could identify additional mutator loci in the BXD. 

### Evidence of epistasis between germline mutator alleles
