## Supplementary information

### Chi-square statistics as an alternative to cosine distance in the inter-haplotype distance method 

Although the current implementation of inter-haplotype distance (IHD) uses the cosine distance metric to compare aggregate mutation spectra at biallelic markers, we also explored the use of chi-square statistics to compare spectra.

As described in the *Materials and Methods*, we compute aggregate mutation spectra on haplotypes that inherited either parental allele at each biallelic marker.
Rather than compute the cosine distance between those spectra, we can create a contingency table of size $(2, \ 6 \times 4^{k - 1})$ (or size $(2, \ 6 \times 4^{k - 1} + 1)$ if $k = 1$ and we're including CpG>TpG) that contains the aggregate $k$-mer mutation spectrum on haplotypes that inherited either parental allele. 
Using the contingency table, we then calculate a $\chi^{2}$ test statistic; larger values of the $\chi^{2}$ statistic suggest more "distance" between the aggregate mutation spectra on the two haplotypes.

We discovered the BXD mutator loci on chromosome 4 and 6 using either cosine distances or chi-square statistics to compare mutation spectra; however, using simulated data we found that the cosine distance metric was more sensitive for mutator allele detection across a broad range of parameter choices (Figure @fig:cosine_vs_chisquare).  

### Supplementary Figures

![
**Simulations to assess the power of the inter-haplotype distance method.** 
In each of 100 trials, we simulated genotypes at 1,000 biallelic loci on a toy population of either 50 or 100 haplotypes as follows.
At every locus on every haplotype, we drew a single floating point value from a uniform distribution $[0, 1)$.
If that value was less than or equal to 0.5, we set the allele to be "A"; otherwise, we set the allele to be "B".
In each trial, we also simulated *de novo* germline mutations on the population of haplotypes, such that at a single locus $g_i$, we augmented the mutation rate of a particular $k$-mer by the specified effect size (an effect size of 1.5 indicates a 50% increase in the mutation rate) on haplotypes carrying "A" alleles.
We then applied the inter-haplotype distance method to these simulated data and asked if the adjusted cosine distance at locus $g_i$ was greater than expected by chance. 
Given a specific combination of parameters, the y-axis denotes the fraction of 100 trials in which the simulated mutator allele could be detected at a significance threshold of p = 0.05.
Shaded areas indicate the standard deviation of that fraction across 100 simulations.
](images/fig-power-simulations.png){#fig:simulations tag="1-figure supplement 1" width=8.5in} 

![
**Comparing power between the inter-haplotype distance method and QTL mapping.** 
In each of 100 trials, we simulated genotypes at 1,000 biallelic loci on a toy population of 50 haplotypes as follows.
At every locus on every haplotype, we drew a single floating point value from a uniform distribution $[0, 1)$.
If that value was less than or equal to 0.5, we set the allele to be "A"; otherwise, we set the allele to be "B".
In each trial, we also simulated *de novo* germline mutations on the population of haplotypes, such that at a single locus $g_i$, we augmented the rate of the specified mutation type by the specified effect size (an effect size of 1.5 indicates a 50% increase in the mutation rate) on haplotypes carrying "A" alleles.
We then applied the inter-haplotype distance method to these simulated data and asked if the adjusted cosine distance at locus $g_i$ was greater than expected by chance. 
Similarly, in each trial, we used R/qtl2 to perform a genome scan for QTL and asked if the log-odds score at $g_i$ was greater than expected by chance.
Given a specific combination of parameters, the y-axis denotes the fraction of 100 trials in which the simulated mutator allele could be detected at a significance threshold of p = 0.05 (for IHD) or at an alpha of $\frac{0.05}{7}$ (for QTL mapping).
Shaded areas indicate the standard deviation of that fraction across 100 simulations.
](images/fig-power-comparison.png){#fig:ihd_vs_qtl_power tag="1-figure supplement 2" width=8.5in} 

![
**Comparing the power of cosine distance and chi-square statistics in the inter-haplotype distance method.** 
In each of 100 trials, we simulated genotypes at 1,000 biallelic loci on a toy population of 50 haplotypes as follows.
At every locus on every haplotype, we drew a single floating point value from a uniform distribution $[0, 1)$.
If that value was less than or equal to 0.5, we set the allele to be "A"; otherwise, we set the allele to be "B".
In each trial, we also simulated *de novo* germline mutations on the population of haplotypes, such that at a single locus $g_i$, we augmented the rate of the specified mutation type by the specified effect size (an effect size of 1.5 indicates a 50% increase in the mutation rate) on haplotypes carrying "A" alleles.
We then applied the inter-haplotype distance method, using either cosine distance or chi-square statistics to compare aggregate mutation spectra at each marker.
In each trial, we asked if the adjusted cosine distance or chi-square statistic at locus $g_i$ was greater than expected by chance. 
Given a specific combination of parameters, the y-axis denotes the fraction of 100 trials in which the simulated mutator allele could be detected at a significance threshold of p = 0.05.
Shaded areas indicate the standard deviation of that fraction across 100 simulations.
](images/fig-distance-method-comparison.png){#fig:cosine_vs_chisquare tag="1-figure supplement 3" width=8.5in} 

![
**Results of inter-haplotype distance scans using BXDs with *B* alleles at the chromosome 4 locus.** 
Adjusted cosine distances between aggregate 1-mer *de novo* mutation spectra on BXD haplotypes with *B* alleles at `rs52263933` (n = 38 haplotypes; 22,080 total mutations) and either *D* or *B* alleles at 7,278 informative markers. 
Cosine distance threshold at p = 0.05 was calculated by performing 10,000 permutations of the BXD haplotype mutation data, and is shown as a dotted grey line.
](images/fig-distance-results-conditioned.png){#fig:distance-results-conditioned tag="2-figure supplement 1" width=8.5in} 

![
**Quantitative trait locus scans for mutation spectrum phenotypes.**
Using the BXDs with D genotypes at `rs52263933` (the marker with the highest cosine distance on chromosome 4; n = 55 RILs, 40,913 total mutations), we used R/qtl2 to perform QTL scans for the centered log-ratio (CLR) transformed fraction of each 1-mer mutation type.
QTL scans also included a kinship matrix (that contained the pairwise genetic similarity between each pair of BXDs, calculated using the leave-one-chromosome-out method) as a random effect term using the `kinship` keyword argument in the `scan1` function.
Plots show the log-odds (LOD) score at every genotyped marker in blue; the dotted black line represents the genome-wide LOD significance threshold (established using 1,000 permutations at an alpha of $\frac{0.05}{7}$ to account for the fact that 7 separate association tests were performed.)
](images/fig-qtl-scans.png){#fig:qtl-scans tag="2-figure supplement 2" width=8.5in} 

![
**Mutation spectra comparison in BXD strains.**
Fractions of *de novo* germline mutations in BXDs with either *D* or *B* genotypes at markers `rs52263933` and `rs31001331`, stratified by mutation type.
](images/fig-spectra-comparison-bxd-all.png){#fig:spectra-comparison-all tag="3-figure supplement 1" width=8.5in}

![
**Mutation spectra comparison in Sanger Mouse Genomes Project strains.**
Fractions of *de novo* germline mutations in Sanger MGP strains with either *D* or *B* haplotypes at the chromosome 4 and chromosome 6 mutator loci, stratified by mutation type.
](images/fig-spectra-comparison-mgp.png){#fig:spectra-comparison-mgp tag="3â€”figure supplement 2" width=8.5in} 

### Supplementary Tables

| Gene name | Tissue name | # BXDs with expression data |  Top significant marker | LRS at top significant marker | Significant LRS threshold | Additive effect of D allele on expression |
| - | - | - | - | - | - |
| *Ogg1* | Kidney | 53 | `rsm10000004188` | 52.25 | 17.82 | -0.186 |
| *Ogg1* | Gastrointestinal | 46 | `rsm10000003441` | 23.39 | 16.09 | -0.074 |
| *Ogg1* | Hematopoetic stem cells | 22 | - | - | 16.45 | - | 
| *Ogg1* | Hematopoetic progenitor cells | 23 | - | - | 18.52 | - | 
| *Ogg1* | Spleen | 79 | `rsm10000003418` | - | 17.51 | - | 
| *Ogg1* | Liver | 50 | `rsm10000004188` | 53.54 | 18.77 | -0.156 | 
| *Ogg1* | Heart | 73 | - | - | 16.22 | - |
| *Ogg1* | Eye | 87 | `rsm10000004194` | 23.05 | 16.96 | 0.088 |  
| *Mbd4* | Kidney | 53 | - | - | 18.48 | - |
| *Mbd4* | Gastrointestinal | 46 | - | - | 15.97 | - |
| *Mbd4* | Hematopoetic stem cells | 22 | - | - | 18.12 | - | 
| *Mbd4* | Hematopoetic progenitor cells | 23 | - | - | 18.55 | - | 
| *Mbd4* | Spleen | 79 | `rsm10000004199` | 21.42 | 16.99 | 0.069 | 
| *Mbd4* | Liver | 50 | - | - | 16.15 | - | 
| *Mbd4* |Heart | 73 | - | - | 17.17 | - |
| *Mbd4* | Eye | 87 | - | - | 16.63 | - |  

Table: Presence or absence of cis-eQTLs for *Ogg1* and *Mbd4* in various tissues identified using GeneNetwork. {#tbl:eqtl-results tag="supplement 1"}

| SV start | SV end |  SV type | Gene name(s) | Overlaps exon? |
| - | - | - | - | - |
| 108,437,600 | 108,437,652 | DEL | *Gm35165*, *Itpr1* | Yes | 
| 110,617,754 | 110,618,347 | DEL | *Gm20387* | Yes | 
| 113,110,214 | 113,110,215 | INS | *Setd5* | Yes | 
| 116,217,366 | 116,217,367 | INS | *Gm52873*, *Washc2* | Yes |
| 116,323,201 | 116,323,202 | INS | *Marchf8* | Yes | 
| 116,662,107 | 116,668,530 | DEL | *Tmem72* | Yes | 
| 116,976,917 | 116,976,984 | DEL | *LOC118567448* | Yes | 
| 112,605,594 | 112,605,595 | INS | *Rad18* | No | 
| 112,629,618 | 112,636,619 | DEL | *Rad18* | No | 
| 115,823,860 | 115,824,076 | DEL | *Mbd4* | No |

Table: Large structural variants overlapping protein-coding genes in the mutator locus on chromosome 6 All coordinates are with respect to GRCm39/mm39. {#tbl:sv-overlap tag="supplement 2"}

