## Supplementary information

### Using cosine distance instead of chi-square statistics for comparing mutation spectra

We also explored the use of cosine distance as an alternative to the $\chi^{2}$ statistic for comparing mutation spectra in our inter-haplotype distance scans. 
The cosine distance between two vectors $\mathbf{A}$ and $\mathbf{B}$ is defined as 

$$D_C = 1 - \frac{\mathbf{A} \cdot \mathbf{B}}{||\mathbf{A}|| \ ||\mathbf{B}||}$$

where $||\mathbf{A}||$ and $||\mathbf{B}||$ are the $L^2$ (or Euclidean) norms of $\mathbf{A}$ and $\mathbf{B}$, respectively. 
The cosine distance metric has a number of favorable properties for comparing mutation spectra. 
Since cosine distance does not take the magnitude of vectors into account, it can be used to compare two spectra with unequal total mutation counts (even if those total counts are relatively small). 
Additionally, by calculating the cosine distance between mutation *spectra*, we avoid the need to perform separate comparisons of mutation counts at each individual $k$-mer mutation type. 

In practice, we found that the $\chi^2$ statistic was more sensitive for detecting loci associated with differences in mutation spectra. 
However, we provide the ability to use cosine distance in our method, as well, since the $\chi^2$ statistic may not behave as expected in certain situations (e.g., if the counts of mutations in each $k$-mer type are small).


### Supplementary Figures

![
**Simulations to assess the power of the inter-haplotype distance method.** 
In each of 100 trials, we simulated genotypes at 1,000 biallelic loci on a toy population of either 50 or 100 haplotypes as follows.
At every locus on every haplotype, we drew a single floating point value from a uniform distribution ($[0, 1)$).
If that value was less than or equal to 0.5, we set the allele to be "A"; otherwise, we set the allele to be "B".
In each trial, we also simulated *de novo* germline mutations on the population of haplotypes, such that at a single locus $g_i$, we augmented the mutation rate of a particular $k$-mer by the specified effect size (an effect size of 1.5 indicates a 50% increase in the mutation rate) on haplotypes carrying "A" alleles.
We then applied the inter-haplotype distance method to these simulated data and asked if the adjusted $\chi^2$ statistic at locus $g_i$ was greater than expected by chance. 
Given a specific combination of parameters, the y-axis denotes the fraction of 100 trials in which the simulated mutator allele could be detected at a significance threshold of p = 0.05.
Shaded areas indicate the standard deviation of that fraction across 100 simulations.
](images/fig-power-simulations.png){#fig:simulations tag="1-figure supplement 1" width=7in} 

![
**Comparing power between the inter-haplotype distance method and QTL mapping.** 
In each of 100 trials, we simulated genotypes at 1,000 biallelic loci on a toy population of either 50 or 100 haplotypes as follows.
At every locus on every haplotype, we drew a single floating point value from a uniform distribution ($[0, 1)$).
If that value was less than or equal to 0.5, we set the allele to be "A"; otherwise, we set the allele to be "B".
In each trial, we also simulated *de novo* germline mutations on the population of haplotypes, such that at a single locus $g_i$, we augmented the C>A mutation rate by the specified effect size (an effect size of 1.5 indicates a 50% increase in the mutation rate) on haplotypes carrying "A" alleles.
We then applied the inter-haplotype distance method to these simulated data and asked if the adjusted $\chi^2$ statistic at locus $g_i$ was greater than expected by chance. 
Similarly, in each trial, we used R/qtl2 to perform a genome scan for QTL and asked if the log-odds score at $g_i$ was greater than expected by chance.
Given a specific combination of parameters, the y-axis denotes the fraction of 100 trials in which the simulated mutator allele could be detected at a significance threshold of p = 0.05.
Shaded areas indicate the standard deviation of that fraction across 100 simulations.
](images/fig-power-comparison.png){#fig:ihd_vs_qtl_power tag="1-figure supplement 2" width=7in} 

![
**Results of inter-haplotype distance scans in the BXD RILs using 3-mer mutation spectra.** 
**a)** Adjusted $\chi^{2}$ statistics between aggregate 3-mer *de novo* mutation spectra on BXD haplotypes (n = 93 haplotypes; 62,993 total mutations) with either *D* or *B* alleles at 7,320 informative markers. 
Distance threshold at $p = 0.05$ was calculated by performing 10,000 permutations of the BXD haplotype mutation data, and is shown as a dotted grey line. 
**b)** Adjusted $\chi^{2}$ statistics between aggregate 3-mer *de novo* mutation spectra on BXD haplotypes with *D* alleles at `rs52263933` (n = 55 haplotypes; 40,913 total mutations) and either *D* or *B* alleles at 7,320 informative markers. 
Distance threshold at $p = 0.05$ was calculated by performing 10,000 permutations of the BXD haplotype mutation data, and is shown as a dotted grey line.
](images/fig-distance-3mer.png){#fig:distance-3mer tag="2—figure supplement 1" width=7in} 


![
**Mutation spectra comparison in Sanger Mouse Genomes Project strains.**
Fractions of *de novo* germline mutations in Sanger MGP strains with either *D* or *B* genotypes at markers `rs52263933` and `rs31001331`, stratified by mutation type.
](images/fig-spectra-comparison-mgp.png){#fig:spectra-comparison-mgp tag="3—figure supplement 1" width=7in} 

### Supplementary Tables

| Gene name | Tissue name | # BXDs with expression data |  Top significant marker | LRS at top significant marker | Significant LRS threshold | Additive effect of D allele on expression |
| - | - | - | - | - | - |
| *Ogg1* | Kidney | 53 | `rsm10000004188` | 52.25 | 17.82 | -0.186 |
| *Ogg1* | Gastrointestinal | 46 | `rsm10000003441` | 23.39 | 16.09 | -0.074 |
| *Ogg1* | Hematopoetic stem cells | 22 | - | - | 16.45 | - | 
| *Ogg1* | Hematopoetic progenitor cells | 23 | - | - | 18.52 | - | 
| *Ogg1* | Spleen | 79 | `rsm10000003418` | - | 17.51 | - | 
| *Ogg1* | Liver | 50 | `rsm10000004188` | 53.54 | 18.77 | -0.156 | 
| *Ogg1* | Heart | 73 | - | - | 16.22 | - |
| *Ogg1* | Eye | 87 | `rsm10000004194` | 23.05 | 16.96 | 0.088 |  
| *Mbd4* | Kidney | 53 | - | - | 18.48 | - |
| *Mbd4* | Gastrointestinal | 46 | - | - | 15.97 | - |
| *Mbd4* | Hematopoetic stem cells | 22 | - | - | 18.12 | - | 
| *Mbd4* | Hematopoetic progenitor cells | 23 | - | - | 18.55 | - | 
| *Mbd4* | Spleen | 79 | `rsm10000004199` | 21.42 | 16.99 | 0.069 | 
| *Mbd4* | Liver | 50 | - | - | 16.15 | - | 
| *Mbd4* |Heart | 73 | - | - | 17.17 | - |
| *Mbd4* | Eye | 87 | - | - | 16.63 | - |  

Table: Presence or absence of cis-eQTLs for *Ogg1* and *Mbd4* in various tissues identified using GeneNetwork. {#tbl:eqtl-results tag="supplement 1"}

